### PR TITLE
libmodbus: new package

### PR DESCRIPTION
**Pull request by shaynakapadia**
Adding the libmodbus package to spack

### CODE REVIEW

File: var/spack/repos/builtin/packages/libmodbus/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+#     spack install libmodbus
+#
+# You can edit this file again by typing:
+#
+#     spack edit libmodbus
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmodbus/package.py]: You can delete this stuff, it's only there for new users to create their first package.

File: var/spack/repos/builtin/packages/libmodbus/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from spack import *
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmodbus/package.py:7]: Flake8 will complain and say it wants 2 blank lines here.

File: var/spack/repos/builtin/packages/libmodbus/package.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    version('3.1.2', sha256='661e14f9dc904f3f1b034464ddaa5fd4b8472f8f5d1ea10a1148af85591b7ee9')
+    version('3.1.1', sha256='76d93aff749d6029f81dcf1fb3fd6abe10c9b48d376f3a03a4f41c5197c95c99')
+    version('3.0.8', sha256='022f0691d920b8aee3ee49d7af0f69b7ef80fc3c849a8e0281d5bc27db7a24ea')
+    version('3.0.7', sha256='6c26850cd5dedcf5dad40977ac7f2ee990a3667f6959a1e05e22959bdf537961')
+    version('3.0.6', sha256='046d63f10f755e2160dc56ef681e5f5ad3862a57c1955fd82e0ce036b69471b6')
+    version('3.0.5', sha256='19aad5d55fa315602d6e836a858a3802f1608f9d824afba05fa12a58a1b1e656')
+
+    depends_on('autoconf')
+    depends_on('automake')
+    depends_on('libtool')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmodbus/package.py]: I believe these dependencies are only needed when your tarball doesn't contain a `configure` script, but yours does, so these shouldn't be needed.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Hi @shaynakapadia, welcome to Spack! Just a few minor suggestions before your PR is good to go!

File: var/spack/repos/builtin/packages/libmodbus/package.py (hunk 1 above)
Comment (shaynakapadia) [reply var/spack/repos/builtin/packages/libmodbus/package.py]: Oh right did not see that, thanks!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**